src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java

```diff
@@ -78,19 +71,40 @@
     private List<String> runtimeArgsList;
     private List<String> releaseSymbolsList;
 
-    private String appName;
-    private String mainClassName;
-
     private IosSigningConfiguration iosSigningConfiguration = new IosSigningConfiguration();
 
-    public ProjectConfiguration() {}
+    private ProjectConfiguration publicConfig;
+
+    /**
+     * Private projects configuration, which includes everything, including public settings
+     * @param config public project configuration
+     */
+    public InternalProjectConfiguration(ProjectConfiguration config ) {
+
+        this.publicConfig = Objects.requireNonNull(config);
+
+        boolean usePrismSW = Boolean.parseBoolean(System.getProperty("prism.sw", "false"));
+        boolean skipCompile = Boolean.parseBoolean(System.getProperty("skipcompile", "false"));
+        boolean skipSigning = Boolean.parseBoolean(System.getProperty("skipsigning", "false"));
+
+        setUsePrismSW(usePrismSW);
+        getIosSigningConfiguration().setSkipSigning(skipSigning);
 
-    public String getGraalPath() {
-        return this.graalPath;
+        setJavaStaticLibs(System.getProperty("javalibspath")); // this can be safely set even if null. Default will be used in that case
+        setJavaFXStaticSDK(System.getProperty("javafxsdk"));  // this can be safely set even if null. Default will be used in that case
+
+        setReflectionList(splitString(System.getProperty("reflectionlist")));
```

Comment:
In `SubstrateDispatcher::main`, `reflectionList` (and other lists), come from a System property, however, when we use the plugin, the plugin itself will create an instance of `ProjectConfiguration`, will call the setters, so in this case we should use here `setReflectionList(config.getReflectionList())`.